hugot_xla_test.go

```diff
@@ -230,3 +234,73 @@ func TestThreadSafetyXLACuda(t *testing.T) {
 	}(session)
 	threadSafety(t, session, 1000)
 }
+
+// TEMP: testing how we can do inference with XLA and gemma
+
+type ModelConfig struct {
+	NumKeyValueHeads int   `json:"num_key_value_heads"`
+	HeadDim          int   `json:"head_dim"`
+	NumHiddenLayers  int   `json:"num_hidden_layers"`
+	EosTokenID       []int `json:"eos_token_id"`
+}
+
+func CreateCache(batchSize, numLayers, numKeyValueHeads, seqLen, headDim int) []*tensors.Tensor {
+	cache := make([]*tensors.Tensor, numLayers*2)
+
+	for layer := 0; layer < numLayers; layer++ {
+		keyTensor := tensors.FromScalarAndDimensions(float32(0), batchSize, numKeyValueHeads, seqLen, headDim)
+		cache[layer*2] = keyTensor
+
+		valueTensor := tensors.FromScalarAndDimensions(float32(0), batchSize, numKeyValueHeads, seqLen, headDim)
+		cache[layer*2+1] = valueTensor
+	}
+	return cache
+}
+
+func check(err error) {
+	if err != nil {
+		panic(err.Error())
+	}
+}
+
+func TestHugoPipeline(t *testing.T) {
+	session, err := NewXLASession()
+	check(err)
+
+	defer func(session *Session) {
+		err := session.Destroy()
+		check(err)
+	}(session)
+
+	config := TextGenerationConfig{
+		ModelPath:    "/home/testuser/repositories/onnx_models",
+		Name:         "test pipeline",
+		OnnxFilename: "gemma_model.onnx",
+		Options: []pipelineBackends.PipelineOption[*pipelines.TextGenerationPipeline]{
+			pipelines.WithMaxTokens(15),
+		},
+	}
+
+	gemmaPipeline, err := NewPipeline(session, config)
+
+	check(err)
+
+	batch := []string{"what is the capital of the Netherlands?",
+		"who was the first president of the United States?"}
+	// batch := []string{"what is the capital of the Netherlands?"}
+
+	batchResult, err := gemmaPipeline.Run(batch)
+	check(err)
+
+	fmt.Println(batchResult.GetOutput())
+}
+
+func TestTextGenerationPipelineTokensXLA(t *testing.T) {
```

Comment:
rename to TestTextGenerationPipeline, copy the structure of the other tests, also move it to be with the other pipeline tests in this file.  Add TestTextGenerationPipelineValidation test to match the other pipelines. Add these two tests to hugot_go_test also